cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_rfc_queue.h

```diff
@@ -0,0 +1,106 @@
+ */
+
+#ifndef CC26XX_CC13XX_RFC_QUEUE_H
+#define CC26XX_CC13XX_RFC_QUEUE_H
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Data queue
+ */
+typedef struct {
+    uint8_t *curr_entry; /**< Pointer to the data entry to be used */
+    uint8_t *last_entry; /**< Last entry of the queue, NULL is queue is
+                              circular */
+} rfc_data_queue_t;
+
+/**
+ * @brief   Data entry status
+ */
+typedef enum {
+    RFC_DATA_ENTRY_PENDING = 0,   /**< Entry not yet used */
+    RFC_DATA_ENTRY_ACTIVE = 1,    /**< Entry in use by radio CPU */
+    RFC_DATA_ENTRY_BUSY = 2,      /**< Entry being updated */
+    RFC_DATA_ENTRY_FINISHED = 3,  /**< Radio CPU is finished accessing the
+                                       entry */
+    RFC_DATA_ENTRY_UNFINISHED = 4 /**< Radio CPU is finished accessing the
+                                       entry, but packet could not be
+                                       finished */
+} rfc_data_entry_status_t;
+
+/**
+ * @brief   Data entry type
+ */
+typedef enum {
+    RFC_DATA_ENTRY_TYPE_GEN  = 0,   /**< General type: Tx entry or single
+                                         element Rx entry */
+    RFC_DATA_ENTRY_TYPE_MULTI = 1,  /**< Multi-element Rx entry type */
+    RFC_DATA_ENTRY_TYPE_PTR = 2,    /**< Pointer entry type */
+    RFC_DATA_ENTRY_TYPE_PARTIAL = 3 /**< Partial read entry type */
+} rfc_data_entry_type_t;
+
+/**
+ * @brief   Data entry configuration
+ */
+typedef struct {
+    uint8_t type:2; /**< Entry type */
+    uint8_t lensz:2; /**< Length size for this entry */
+    uint8_t irq_intv:4; /**< Reserved */
+} rfc_data_entry_config_t;
+
+/**
+ * @brief   Data entry
+ */
+typedef struct {
+   uint8_t* next_entry; /**< Pointer to next entry in the queue */
+   uint8_t status; /**< Indicates status of entry, including whether it is free
+                        for the system CPU to write to */
+   rfc_data_entry_config_t config; /**< Data entry configuration */
+   uint16_t length; /**< For pointer entries: Number of bytes in the data buffer
+                         pointed to. For other entries: Number of bytes
+                         following this length field */
+} rfc_data_entry_t;
+
+/**
+ * @brief   Data entry general
+ */
+typedef struct {
+   uint8_t* next_entry; /**< Pointer to next entry in the queue */
+   uint8_t status; /**< Indicates status of entry, including whether it is free
+                        for the system CPU to write to */
+   rfc_data_entry_config_t config; /**< Data entry configuration */
+   uint16_t length; /**< For pointer entries: Number of bytes in the data buffer
+                         pointed to. For other entries: Number of bytes
+                         following this length field */
+   uint8_t data; /**< First byte of the data array to be received or
+                      transmitted */
+} rfc_data_entry_general_t;
```

Comment:
I have a question about radio_general_data_entry_t structures. I have been observing in the datasheet that the last data element referred to in the code is taking a `uint8_t` value, my question is the following, according to what is specified in table 25-11, the data member should not be a type pointer `void`? (I mean void because it is a content of any type and its size is unknown)
![image](https://user-images.githubusercontent.com/93989980/198723278-d13e25e1-aab7-47ae-ad48-7310f20c176c.png)

Reply:
It's an `uint8_t*` pointer because the data it points to is a buffer which happens to contain that structure/header at the start, a `void*` works too but it's semantics IMO.

Reply:
Well maybe the `void*` would solve the casting problem? This warnings were activated on the latest release so I really don't know what was the brackground. E.g.:

```gcc
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/radio/cc26x2_cc13x2_rf_internal.c: In function 'cc26x2_cc13x2_rf_internal_init':
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/radio/cc26x2_cc13x2_rf_internal.c:467:35: warning: cast increases required alignment of target type [-Wcast-align]
  467 |         rfc_data_entry_t *entry = (rfc_data_entry_t *)_rx_buf[i];
      |                                   ^
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/radio/cc26x2_cc13x2_rf_internal.c: In function 'cc26x2_cc13x2_rf_recv':
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/radio/cc26x2_cc13x2_rf_internal.c:623:12: warning: cast increases required alignment of target type [-Wcast-align]
  623 |     return (rfc_data_entry_general_t *)cc26x2_cc13x2_rfc_queue_recv(&_rx_queue);
      |            ^
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/radio/cc26x2_cc13x2_rf_radio_ops.c: In function 'cc26x2_cc13x2_rfc_isr':
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/radio/cc26x2_cc13x2_rf_radio_ops.c:124:28: warning: cast increases required alignment of target type [-Wcast-align]
  124 |         uint16_t pktlen = *(uint16_t *)&entry->data;
      |                            ^
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/radio/cc26x2_cc13x2_rf_radio_ops.c: In function '_read':
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/radio/cc26x2_cc13x2_rf_radio_ops.c:239:15: warning: cast increases required alignment of target type [-Wcast-align]
  239 |     pktlen = *(uint16_t *)&entry->data;
      |               ^
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/radio/cc26x2_cc13x2_rf_radio_ops.c: In function '_len':
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/radio/cc26x2_cc13x2_rf_radio_ops.c:277:24: warning: cast increases required alignment of target type [-Wcast-align]
  277 |     uint16_t pktlen = *(uint16_t *)&entry->data;
      |                        ^
"make" -C /Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/rfc
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/rfc/cc26x2_cc13x2_rfc_queue.c: In function 'cc26x2_cc13x2_rfc_queue_recv':
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/rfc/cc26x2_cc13x2_rfc_queue.c:26:37: warning: cast increases required alignment of target type [-Wcast-align]
   26 |     rfc_data_entry_t *start_entry = (rfc_data_entry_t *)queue->curr_entry;
      |                                     ^
/Users/facosta/git/RIOT-OS/RIOT/cpu/cc26x2_cc13x2/rfc/cc26x2_cc13x2_rfc_queue.c:34:22: warning: cast increases required alignment of target type [-Wcast-align]
   34 |         curr_entry = (rfc_data_entry_t *)(curr_entry->next_entry);
      |
```